Prometheus: Browser resource caching (#60711)
grafana/grafana: docs/sources/administration/provisioning/index.md

| prometheusVersion          | string  | Prometheus                                                       | The version of the Prometheus data source, such as `2.37.0`, `2.24.0`                                                                                                                                                                                                                                               |
| prometheusType             | string  | Prometheus                                                       | The type of the Prometheus data sources. such as `Prometheus`, `Cortex`, `Thanos`, `Mimir`                                                                                                                                                                                                                          |

> | cacheLevel                 | string  | Prometheus                                                       | The setting that determines how aggressive the browser cache is, valid values are: `Low`, `Medium`, `High`, or `None`. This field is only configurable when the prometheusResourceBrowserCache feature flag is enabled.                                                                                             |

Reviewer: ```suggestion
| cacheLevel                 | string  | Prometheus                                                       | This determines how aggressive the browser cache is. Valid values include: `Low`, `Medium`, `High`, and `None`. This field is configurable when you enable the `prometheusResourceBrowserCache` feature flag.                                                                                             |
```

> | cacheLevel                 | string  | Prometheus                                                       | The setting that determines how aggressive the browser cache is, valid values are: `Low`, `Medium`, `High`, or `None`. This field is only configurable when the prometheusResourceBrowserCache feature flag is enabled.                                                                                             |

Reviewer: What do you mean by 'aggressive'?

Author: I removed the word "aggressive", basically this feature does a host of cache related things, including increasing debounce intervals, setting the cache duration for metadata, and setting the cache duration for other resource calls.

Author: Should we document what each of the settings do? Aka a setting of Low will give a debounce interval of 350ms, cache metadata for 1 day, and cache all other resource calls made within the same minute. Medium will me 600ms debounce, metadata for 1 week, and all resource queries made within 10 minutes...

Author: The only concern I have with adding that is that the code will change and folks won't change the docs, should we link to the code? Or is that too low level and we want some additional high level description?

Reviewer: I'm unfamiliar with linking the docs to the code. Do you mean auto-populating the docs based on the code? And looking into this further, we want to update the product docs to account for this change given its importance. Will this be the default experience at some point? I'm not sure exactly which topic to update, perhaps this one? https://grafana.com/docs/grafana/v9.4/datasources/prometheus/query-editor/

I do feel like we should describe the benefits and promote its use. Thoughts?

Author: This will potentially be the default experience in the future, but for now we're launching behind a feature flag so we can get this out to a few customers to help test this, so maybe we can worry about the more public announcement documentation (it's also been suggested that we write a blog post about this) when we're ready to remove this feature flag and make this the default behavior?

Reviewer: That works for me, thank you!
